File: backend/src/main/java/com/xx/music/model/entity/Song.java
package com.xx.music.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "t_song")
public class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "song_id")
    private String songId;

    @Column(name = "name")
    private String name;

    @Column(name = "default_artist_id")
    private Long defaultArtistId;

    @Column(name = "album_id")
    private Long albumId;

    @Column(name = "duration")
    private Integer duration;

    @Column(name = "is_vip")
    private Integer isVip;

    @Column(name = "has_standard")
    private Integer hasStandard;

    @Column(name = "has_high")
    private Integer hasHigh;

    @Column(name = "has_lossless")
    private Integer hasLossless;

    @Column(name = "lyric_url")
    private String lyricUrl;

    @Column(name = "tlyric_url")
    private String tlyricUrl;

    @Column(name = "play_count")
    private Long playCount;

    @Column(name = "status")
    private Integer status;

    @Column(name = "source")
    private String source;

    @Column(name = "source_id")
    private String sourceId;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: admin-frontend/src/api/song.ts
import request from '@/utils/request'

export function getSongs(params: {
  page?: number
  size?: number
  keyword?: string
  status?: number
  isVip?: number
  artistId?: string
  albumId?: string
  sortBy?: string
  sortOrder?: string
}) {
  return request.get('/songs', { params })
}

export function getSongDetail(songId: string) {
  return request.get(`/songs/${songId}`)
}

export function createSong(data: {
  name: string
  artistIds: string[]
  albumId?: string
  duration: number
  isVip?: boolean
  hasStandard?: boolean
  hasHigh?: boolean
  hasLossless?: boolean
  lyricUrl?: string
  tlyricUrl?: string
  source: string
  sourceId: string
  status?: number
}) {
  return request.post('/songs', data)
}

export function updateSong(songId: string, data: {
  name?: string
  artistIds?: string[]
  albumId?: string
  duration?: number
  isVip?: boolean
  hasStandard?: boolean
  hasHigh?: boolean
  hasLossless?: boolean
  lyricUrl?: string
  tlyricUrl?: string
  source?: string
  sourceId?: string
}) {
  return request.put(`/songs/${songId}`, data)
}

export function updateSongStatus(songId: string, status: number) {
  return request.put(`/songs/${songId}/status`, { status })
}

export function deleteSong(songId: string) {
  return request.delete(`/songs/${songId}`)
}
